contracts/actions/liquityV2/trove/LiquityV2Open.sol

```diff
@@ -0,0 +1,151 @@
+        Params memory params = parseInputs(_callData);
+        (, bytes memory logData) = _liquityOpen(params);
+        logger.logActionDirectEvent("LiquityV2Open", logData);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            ACTION LOGIC
+    //////////////////////////////////////////////////////////////*/
+    function _liquityOpen(Params memory _params) internal returns (uint256, bytes memory) {
+        address collToken = IAddressesRegistry(_params.market).collToken();
+
+        _pullCollateralAndGasCompensation(_params, collToken);
+
+        address borrowerOperations = IAddressesRegistry(_params.market).borrowerOperations();
+
+        _approveCollateralAndGasCompensation(_params, collToken, borrowerOperations);
+
+        if (_params.interestBatchManager != address(0)) {
+            IBorrowerOperations(borrowerOperations).openTroveAndJoinInterestBatchManager(
+                IBorrowerOperations.OpenTroveAndJoinInterestBatchManagerParams({
+                    owner: address(this),
+                    ownerIndex: _params.ownerIndex,
+                    collAmount: _params.collAmount,
+                    boldAmount: _params.boldAmount,
+                    upperHint: _params.upperHint,
+                    lowerHint: _params.lowerHint,
+                    interestBatchManager: _params.interestBatchManager,
+                    maxUpfrontFee: _params.maxUpfrontFee,
+                    addManager: address(0),
+                    removeManager: address(0),
+                    receiver: address(0)
+                })
+            );
+        } else {
+            IBorrowerOperations(borrowerOperations).openTrove(
+                address(this),
+                _params.ownerIndex,
+                _params.collAmount,
+                _params.boldAmount,
+                _params.upperHint,
+                _params.lowerHint,
+                _params.annualInterestRate,
+                _params.maxUpfrontFee,
+                address(0),
+                address(0),
+                address(0)
+            );
+        }
+
+        BOLD_ADDR.withdrawTokens(_params.to, _params.boldAmount);
+
+        return (_params.collAmount, abi.encode(_params));
+    }
+
+    function _pullCollateralAndGasCompensation(Params memory _params, address _collToken) internal {
+        if (_collToken == TokenUtils.WETH_ADDR) {
+            bool isMaxPull = _params.collAmount == type(uint256).max;
+
+            // when pulling max amount, we need to leave some WETH for gas compensation
+            if (isMaxPull) {
+                _params.collAmount = _collToken.pullTokensIfNeeded(_params.from, _params.collAmount);
+                if (_params.collAmount <= ETH_GAS_COMPENSATION) {
```

Comment:
Do we need this check as it's going to revert on the underflow either way, also if the amount is slightly bigger it's still going to revert because it's not going to be enough collateral for min_debt.

Reply:
It’s not necessary; it was more of a sanity check to communicate more explicitly to the caller that WETH compensation is required and that the max uint256 will account for this. It’s true that this would need to be significantly higher than just the WETH gas compensation amount to cover the debt on the Liquity side.